design/marketplace.md

```diff
@@ -0,0 +1,219 @@
+storing the data and the collateral that hosts are expected to deposit. It
+contains parameters pertaining to storage proofs and erasure coding. And
+finally, it contains the amount of hosts that are expected to store the content,
+including a small amount of host losses that can be tolerated.
+
+
+    StorageContract
+
+      cid                             # content identifier
+
+      reward                          # tokens payed per second per filled slot
+      collateral                      # amount of collateral required per host
+
+      proof probability               # frequency at which proofs are required
+      proof parameters                # proof of retrievability parameters
+      erasure coding                  # erasure coding parameters
+
+      hosts                           # amount of storage hosts (including loss)
+      loss                            # number of allowed host losses
+
+      slots                           # assigned host slots
+
+      timeout                         # slots need to be filled before timeout
+
+Slots
+-----
+
+Initially all host slots are empty. An empty slot can be filled by anyone by
+submitting a correct storage proof together with collateral.
+
+
+        collateral                              collateral
+          proof    proof          missed          proof                missed
+            |        |              |               |                    |
+            v        v              v               v                    v
+            -------------------------------------------------------------------
+     slot:  |///////////////////////|               |////////////////////|
+            -------------------------------------------------------------------
+                                    |                                    |
+                                    v                                    v
+                                collateral                           collateral
+                                  burned                               burned
+
+
+
+            ---------------- time ---------------->
+
+
+The time interval that a slot is filled by a host determines the host payout;
+for every second of the interval a certain amount of tokens are awarded to the
+host. Hosts that fill a slot are required to submit frequent proofs of storage.
+
+When a proof is missed, the collateral associated with a slot is mostly burned,
+except for a fee for the one who marked the proof as missing. The slot is now
+considered empty again until another host submits a correct proof together with
+collateral.
+
+To incentivize repair, we allow the host that takes over a slot to count the
+interval between the missed proof and the new proof towards its own profits.
+This means that when a proof is missed, the incentive to repair increases over
+time.
+
+The situation where the payout on repair becomes larger than the required
```

Comment:
This might also introduce other undesired behaviors, for example, nodes might artificially wait before filling a slot. I would proposed we avoid this by burning the payout for each missing proof, that way the only way to maximize profit is to occupy the slot as soon as possible.

Reply:
I've been struggling with this a bit. Hosts need to expend extra resources to repair a file (downloading from other hosts and paying for their bandwidth, performing the erasure coding operations), which would indicate that some sort of compensation is required. But that compensation should not become too big to avoid tampering with other incentives.

Could you elaborate on the scenario where nodes artificially wait before filling a slot? I don't see a direct problem there; if they are interested in filling a slot, then it's in their best interest to fill the slot as soon as possible, lest someone else beats them to it.

Reply:
I also agree that in the way this part is currently written, one might think that waiting could be a good strategy to maximize profit, but it is also true that nodes are in competition, so it is in their best interest to fill the slot before others. 
It is clear that data repairing is more costly than initial download so providing a few extra incentive might be fair.